rosetta-server/src/ws.rs

```diff
@@ -0,0 +1,101 @@
+mod config;
+mod jsonrpsee_client;
+mod tungstenite_jsonrpsee;
+
+pub use config::{RpcClientConfig, WsTransportClient};
+pub use jsonrpsee::client_transport::ws::WsHandshakeError;
+use jsonrpsee::core::client::{Client, ClientBuilder};
+pub use jsonrpsee_client::RpcClient;
+pub use tungstenite_jsonrpsee::{TungsteniteClient, WsError};
+
+pub async fn default_client(
+    url: &str,
+    config: Option<RpcClientConfig>,
+) -> Result<RpcClient, WsHandshakeError> {
+    let config = config.unwrap_or_default();
+
+    let client = match config.client {
+        WsTransportClient::Auto => {
+            log::info!("Connecting using Socketto...");
+            match build_socketto_client(url).await {
+                Ok(client) => client,
+                Err(error) => {
+                    log::warn!("Socketto failed: {}", error);
+                    log::info!("Connecting using Tungstenite...");
+                    build_tungstenite_client(url, RpcClientConfig::default()).await?
+                }
+            }
+        }
+        WsTransportClient::Socketto => {
+            let client = build_socketto_client(url).await?;
+            log::info!("Connected to {} using Socketto", url);
+            client
+        }
```

Comment:
do we need to support socketto?

Reply:
I leaved it there because socketto is the official subxt client, I decided to leave it as an option in case we have issues with tungstenite.